### DISCUSSION TITLE
Get all the possible returned fields and types for query fields

### DISCUSSION THREAD

**Discussion by nikero41** [0no-co/gql.tada#381] [Q&A]
Is there a way to get the returned object from a query of the schema, not one I have created?
Use case: We need to create a function that will merge fields. To do so we need to handle all fields that the query can return
It would be great if there was a typescript type that contained these info.

Similar to `introspection_types` object but in typescript

**Comment by kitten** [comment]
I assume you want a TypeScript type for an object type?

What you're basically asking for is impossible. GraphQL Object types **have no canonical JSON shape**. That's because any given object can have a variable set of fields selected, fields can be aliased, fragments can be interpolated (conditionally), and fields may receive arguments, which alter the result.
In other words, only fragments can have a canonical, unambiguous TypeScript type.

Related threads:
- https://discord.com/channels/1082378892523864074/1247901049294164018/1247904914957533215
- https://github.com/0no-co/gql.tada/discussions/233
